acceptance/chaos_test.go

```diff
@@ -113,27 +115,39 @@ func TestChaos(t *testing.T) {
 				return
 			default:
 			}
+			atomic.StoreInt64(&roundCopy, int64(round))
 			nodes := rnd.Perm(num)[:rnd.Intn(num)+1]
-
 			log.Infof("round %d: restarting nodes %v", round, nodes)
 			for i := 0; i < num; i++ {
 				clients[i].Lock()
 			}
-			for i := 0; i < num; i++ {
-				log.Infof("restarting %v", i)
+			for _, i := range nodes {
+				select {
+				case <-stopper:
+					break
+				default:
+				}
+				if time.Now().After(deadline) {
+					break
+				}
+				log.Infof("restarting %d", i)
 				c.Kill(i)
 				c.Restart(i)
 				initClient(i)
+			}
+			for i := 0; i < num; i++ {
 				clients[i].Unlock()
 			}
-			for cur := atomic.LoadInt64(&count); time.Now().Before(deadline) &&
-				atomic.LoadInt64(&count) == cur; time.Sleep(time.Second) {
+			for cur := atomic.LoadInt64(&count); time.Now().Before(deadline) && atomic.LoadInt64(&count) == cur; time.Sleep(time.Second) {
```

Comment:
done.  I really wish it would let me do each of the lines of the for on a separate line.